compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1656,7 +1656,25 @@ OMR::Power::TreeEvaluator::vblendEvaluator(TR::Node *node, TR::CodeGenerator *cg
 TR::Register*
 OMR::Power::TreeEvaluator::vcastEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::passThroughEvaluator(node, cg);
+   TR::DataType srcType  = node->getOpCode().getVectorSourceDataType().getVectorElementType();
+   TR::DataType resType  = node->getOpCode().getVectorResultDataType().getVectorElementType();
+
+   if (srcType.isFloatingPoint() &&
```

Comment:
from instruction selection perspective with regard to the ISA, for floating point operations, we should ignore VMX instructions completely, since VSX provides a strict superset. Also, it is a fact that VMX didn't have double-precision support at all. the `srcType.isFloatingPoint()` testing above includes double-precision too. the VRF choice below is conservatively correct/workable, but it is not aligned with the ISA intention. it would be more desirable to look into where/which-case VSR choice otherwise below failed it. Then, decide how to work it out.

Reply:
Please note that this is the case where we cast FP vector into Int vector. So I think we need to move VSR into VMX so that the parent of that node (some VMX instruction on Int vector) functions correctly.

Reply:
that makes more sense now.